utils/datasets.py

```diff
@@ -291,6 +291,42 @@ def __len__(self):
         return 0  # 1E12 frames = 32 streams at 30 FPS for 30 years
 
 
+class InfiniteDataLoader(torch.utils.data.dataloader.DataLoader):
+    '''
+    Dataloader that reuses workers.
+
+    Uses same syntax as vanilla DataLoader.
+    '''
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        object.__setattr__(self, 'batch_sampler', _RepeatSampler(self.batch_sampler))
+        self.iterator = super().__iter__()
```

Comment:
This line here initializes the dataloader once at first time/epoch.

Reply:
got it!